src/CodeGen_LLVM.cpp

```diff
@@ -1301,6 +1302,19 @@ Expr CodeGen_LLVM::sorted_avg(Expr a, Expr b) {
     return a + (b - a)/2;
 }
 
+// Although these look like intrinsics, they are designed to be
```

Comment:
I think that this logic is the right way to handle lowering of divisions for CodeGen_C and related backends too. That would require making these first-class intrinsics, then lifting out the Halide division -> round_to_zero division logic to some common location.

I hesitate to ask you to expand the scope of this PR, but I think that would be a useful refactoring.